src/categoricalvector.jl

```diff
@@ -0,0 +1,85 @@
+
+export CategoricalVector
```

Comment:
We could choose not to export it? 

There aren't conflicts when you use both packages unless you also use `CategoricalVector`.

The nomenclature used within AxisArrays is `Categorical`, which is how `CategoricalVector` came up.

How about `DiscreteVector`?